### PR TITLE
70708/veteran status pdf download

### PR DESCRIPTION
**Pull request by liztownd**
## Summary

 _(Summarize the changes that have been made to the platform)_
- Add PDF download functionality, eligibility rules for Veteran Status availability, update related unit test

 _(What is the solution, why is this the solution)_
-  Link click opens a new tab with Veteran Status card for download or immediate printing
- On Mobile the download is immediate (no new tab)
- Eligibility rules around discharge status code
- Add `recordEvent` function call to track download clicks in GA

 _(Which team do you work for, does your team own the maintenance of this component?)_
- Veteran Status/DigitalID team

 _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
- Flipper end date Mid-March 2024

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#74260](https://github.com/department-of-veterans-affairs/va.gov-team/issues/74260)
[department-of-veterans-affairs/va.gov-team#70708](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70708)
[department-of-veterans-affairs/va.gov-team#72512](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72512)

- _Link to epic if not included in ticket_
[department-of-veterans-affairs/va.gov-team#68863](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68863)

## Testing done

 _Describe what the old behavior was prior to the change_
- Basic copy and link existed, adding the actual download/print functionality

 _Describe the steps required to verify your changes are working as expected_
- For local development using the mock server to run the FE: no change necessary, mock user updated with required information
- Otherwise logging in with staging user 57 or 63 will show the status
- Ensure feature flipper is set to `true` for `profile_show_proof_of_veteran_status`
- Navigate to `/profile/military-information` page
- Proof of Veteran status should appear on page with download link
- Clicking the link will open a new tab with the Veteran Status card displayed for download or printing

 _Describe the tests completed and the results_
- Manually tested with different discharge codes to ensure eligibility rules were being applied
- Manually tested that the card opens and is downloadable and accessible
- Manually tested for keyboard and screen reader accessibility
- Updated the Proof of Veteran Status unit test to include eligibility rules
- Added template test in `/platform/pdf` directory to verify the card is properly generated with the correct data points
- Added `recordEvent` function call to track download clicks in GA

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

Visually the same, except for the new tab that opens (see screen shot below)

|         | Before | After |
| ------- | ------ | ----- |
| Desktop |     N/A   |   Screenshot 2024-01-26 at 9 17 00 AM    |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*
- New template added to `/platform/pdf/templates` directory, with associated test
- Added optional parameter to (with default value to avoid breaking existing use of) the `generatePdf` function in the `/platform/pdf/index` file to enable opening of the generated PDF in a new tab
- Updated unit test in `/profile/components/proof-of-veteran-status` directory
- Added functionality and a constants file in `/profile/components/proof-of-veteran-status` directory

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- Please note: there is one eslint ignore next line for the href url for the download link with associated explanation of why it was necessary to include
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable) N/A

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/platform/pdf/templates/veteran_status.js [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  // Name
+  const name = doc.struct('H1', () => {
+    doc
+      .font(config.headings.H1.font)
+      .fontSize(config.headings.H1.size)
+      .text(data.details.fullName, 110, 115);
+  });
+
+  wrapper.add(name);
+
```
Comment (hasahmed2) [inline src/platform/pdf/templates/veteran_status.js:105]: Recommend removing these comments

**Review by hasahmed2 — DISMISSED** [review]

File: src/platform/pdf/templates/veteran_status.js (hunk 1 above)
Comment (liztownd) [reply src/platform/pdf/templates/veteran_status.js:103]: udpated

**Review by hasahmed2 — DISMISSED** [review]

**Review by hasahmed2 — APPROVED** [review]

File: src/applications/personalization/profile/components/proof-of-veteran-status/ProofOfVeteranStatus.jsx [hunk 2]
```diff
@@ -1,52 +1,145 @@
...
-              started, download the <strong> VA: Health and Benefits </strong>{' '}
-              mobile app.
-            </>
-          }
-        />
-      </div>
-    </div>
+    <>
+      {militaryInformation.serviceHistory.serviceHistory.length > 0 &&
+      eligibilityMap.includes('Y') ? (
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/proof-of-veteran-status/ProofOfVeteranStatus.jsx:81]: Suggestion: You don't need a ternary with  `: null` and should be able to instead use `&& ( render stuff )`

**Review by adamwhitlock1 — APPROVED** [review]
I'm curious about this method of generating the pdf on the client side instead of the server side. Are there any concerns about someone hypothetically being able to generate their own veteran status card with details that are untrue, or misleading. I know its a long shot, but I'm always wary of doing stuff like this client side.

Otherwise I had one small comment, but its good to go!

File: src/applications/personalization/profile/components/proof-of-veteran-status/ProofOfVeteranStatus.jsx (hunk 2 above)
Comment (liztownd) [reply src/applications/personalization/profile/components/proof-of-veteran-status/ProofOfVeteranStatus.jsx:81]: Our goal is to add some eligibility messaging to the user to explain why they might not be seeing the Veteran Status download (i.e. they have a service history but the `characterOfDischargeCode` is one of the "unknown" codes for example) so if it's ok with you as is I'll leave the ternary for now so it's set up for an alternate.

**Comment by liztownd** [comment]
@adamwhitlock1 I understand your concerns about pdf generation client-side. The data passed into the `generatePdf` function comes from the existing state object, which comes from the APIs. (This package is what's being used currently to generate the pdfs for Medical History and Medication lists.)

**Review by rsmithadhoc — APPROVED** [review]

**Pull request merged**